perceval/converters/myqlm_converter.py

```diff
@@ -0,0 +1,151 @@
+
+        for i, instruction in enumerate(qlmc.iterate_simple()):
+            instruction_name = instruction[0]  # name of the Gate
+            instruction_qbit = instruction[-1]  # tuple with list of qbit positions
+            # information carried by instruction
+            # tuple ('Name', [value of the parameter for gate], [list of qbit positions where gate is applied])
+
+            assert instruction_name in qlmc.gate_set, "cannot convert (%s)" % instruction_name
+            # only gates are converted -> checking if instruction is in gate_set of AQASM
+
+            if len(instruction_qbit) == 1:
+                if instruction_name == "H":
+                    ins = Circuit(2, name='H') // BS.H()
+                elif instruction_name == "PH":
+                    phi = instruction[1][0]  # value of the variable parameter in gate
+                    ins = Circuit(2, name='PS') // (1, PS(phi))  # apply phase shift on 2nd mode
+                else:
+                    gate_id = qlmc.ops[i].gate
+                    gate_matrix = qlmc.gateDic[gate_id].matrix  # gate matrix data
+                    gate_u = circ_to_np(gate_matrix)  # gate matrix to numpy
+                    ins = self._create_one_qubit_gate(gate_u)
+                p.add(instruction_qbit[0]*2, ins.copy())
+            else:
+                # only 2 qubit gates
+                c_idx = instruction_qbit[0] * 2
+                c_data = instruction_qbit[1] * 2
+                c_first = min(c_idx, c_data)  # used in SWAP
+
+                if instruction_name == "CNOT":
+                    cnot_idx += 1
+                    if use_postselection and cnot_idx == n_cnot:
+                        cnot_processor = self._postprocessed_cnot_builder.build()
+                        mode_map = {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
+                    else:
+                        cnot_processor = self._heralded_cnot_builder.build()
+                        mode_map = {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
+                    p.add(mode_map, cnot_processor)
+                elif instruction_name == "CSIGN":
+                    # Controlled Z in myqlm is named CSIGN
+                    cz_processor = self._heralded_cz_builder.build()
+                    mode_map = {c_idx: 0, c_idx + 1: 1, c_data: 2, c_data + 1: 3}
+                    p.add(mode_map, cz_processor)
+                elif instruction_name == "SWAP":
+                    # c_idx and c_data are consecutive - not necessarily ordered
+                    p.add(c_first, comp.PERM([2, 3, 0, 1]))
+                else:
+                    raise RuntimeError("Gate not yet supported: %s" % instruction_name)
+        p.with_input(default_input_state)
+        return p
+
+    def _create_one_qubit_gate(self, u):
+        # universal method, takes in unitary and approximates one using
+        # Frobenius method
+        if abs(u[1, 0]) + abs(u[0, 1]) < 2 * min_precision_gate:
```

Comment:
You should declare `ins` (`ins=None`) before the if statement line 135. (see above the comment for the same variable name)